src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AlignedAssembly.java

```diff
@@ -0,0 +1,312 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.esotericsoftware.kryo.DefaultSerializer;
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.SAMFlag;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.TextCigarCodec;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.bwa.BwaMemAlignment;
+import org.broadinstitute.hellbender.utils.read.CigarUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Holding necessary information about a local assembly for use in SV discovery.
+ */
+@DefaultSerializer(AlignedAssembly.Serializer.class)
+public final class AlignedAssembly {
+
+    public final int assemblyId;
+
+    public final List<AlignedContig> listOfContigsWithItsAlignmentIntervals;
+
+    /**
+     * Each assembled contig should have at least one such accompanying structure, or 0 when it is unmapped.
+     */
+    @DefaultSerializer(AlignmentInterval.Serializer.class)
+    static public final class AlignmentInterval {
```

Comment:
Just moved `AlignedContig` to top-level, `AlignmentInterval`, if top-level, would be the same as `AlignmentRegion`.
I've discussed this with Ted and came to this compromise, which conveys the relation of "assembly is composed of contigs, whose essential information, in the discovery stage as it stands now, is the contig sequence and its (stripped down) alignment information" (A maybe unrelated issue is that we might want to reach out to other information stored in the assembly graph, but that's for later)
@tedsharpe mind chime in?